contrib/win32/win32compat/win32_usertoken_utils.c

```diff
@@ -742,6 +742,17 @@ int lookup_principal_name(const wchar_t * sam_account_name, wchar_t * user_princ
 	if (seperator == NULL)
 		return -1;
 
+	wchar_t computer_name[CNLEN + 1];
```

Comment:
@oldium, the previous fix you made in V8.1 is not the ideal fix.
The ssh connection fails when we have Match block in the sshd_config. Example -
Match Group administrators
	AuthorizedKeysFile __PROGRAMDATA__/ssh/administrators_authorized_keys

When we have a match block in sshd_config, we first get the user groups. We do this by first getting the token by calling LSALogonUser() and then retrieve the group from the token.

Reply:
Right, I was able to reproduce this too with the same lines.

Reply:
Fixed in #469, please have a look there.